cmd/numalign/main.go

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	flag "github.com/spf13/pflag"
+
+	"github.com/openshift-kni/debug-tools/internal/pkg/numalign"
+	"github.com/openshift-kni/debug-tools/internal/pkg/vfs"
+)
+
+func main() {
+	var sleepHoursParam = flag.StringP("sleep-hours", "S", "", "sleep hours once done.")
+	var scriptPathParam = flag.StringP("script-path", "P", "", "save test script to this path.")
+	var procfsRootParam = flag.StringP("procfs", "p", "/proc", "procfs root.")
+	var sysfsRootParam = flag.StringP("sysfs", "s", "/sys", "sysfs root.")
+	var debugParam = flag.BoolP("debug", "D", false, "enable debug mode.")
+	flag.Parse()
+
+	if _, ok := os.LookupEnv("NUMALIGN_DEBUG"); !ok || !*debugParam {
+		log.SetOutput(ioutil.Discard)
+	}
+
+	sleepTime := getSleepTime(*sleepHoursParam)
+	log.Printf("SYS: sleep for %v after the check", sleepTime)
+
+	numaRes, err := numalign.NewResources(vfs.LinuxFS{}, *procfsRootParam, *sysfsRootParam, os.Environ(), flag.Args())
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+
+	res := numaRes.CheckAlignment()
+	fmt.Printf("%s", res.JSON())
```

Comment:
should it be log.Printf?

Reply:
no, because this needs to go undecorated (e.g. not timestamp) and on stdout - not stderr